xbmc/filesystem/CurlFile.cpp

```diff
@@ -603,7 +603,7 @@ void CCurlFile::SetCommonOptions(CReadState* state, bool failOnError /* = true *
 
     const std::string userpass =
       m_proxyuser + std::string(":") + m_proxypassword;
-    if (!userpass.empty())
+    if (userpass != ":")
```

Comment:
`if (StringUtils::StartsWith(userpass, ":") || StringUtils::EndsWith(userpass, ":")) ....`
only one or both empty?

Reply:
The original intent was to signify both were empty. I don’t know if one or the other is valid in curl. Maybe it is.

Reply:
Ok, I've done a bit of reading on this. Turns out that user/pass needs to have both and they also need to be separately url encoded and then joined with a `:`.

So I think we merge this PR as is (as this is already in Matrix) and then do new Matrix PR fixing it up correctly. Checking for both and the URL encoding. Would you like to do that PR @howie-f ?

Reply:
yes, did some reading too, but haven't found anything really usable. we can do as you suggest (yet i don't unterstand what is meant by "separately url encoded", but i'll figure out). will then open the matrix-fix this evening when i'm back home

Reply:
Do you want to approve this one so and we'll put it in?

Separately URL encoded means something like this in pseudocode: `URLEncode(username) + ":" + URLEncode(password)`. Maybe `%3A` substitutions for any `:`'s would also work.